MF-1761 - Improve path parameters naming

ianmuchyri opened the pull request:
### What does this do?
Improve path parameters naming
### Which issue(s) does this PR fix/relate to?
Resolves #1761 

### List any changes that modify/break current functionality
None

### Have you included tests for your changes?
No

### Did you document any new/modified functionality?
No

### Notes

ianmuchyri committed:
Improve path parameters naming

Signed-off-by: ianmuchyri <ianmuchiri8@gmail.com>

rodneyosodo reviewed (changes requested):
Also change:

- bootstrap/api/transport.go
- twins/api/http/transport.go

ianmuchyri committed:
Improve path parameter naming

improve path parameter naming for:
   bootstrap/api/transport.go
   twins/api/http/transport.go

Signed-off-by: ianmuchyri <ianmuchiri8@gmail.com>

ianmuchyri committed:
Change functions to suit updated path params

Duplicated the functions decodeView and decodeListByConnection to
form new functions decodeThingView, decodeChannelView,
decodeThingListByConnection and decodeChannelListByConnection. This
was as a result of the two functions being used for both view thing
and view channel services

Signed-off-by: ianmuchyri <ianmuchiri8@gmail.com>

rodneyosodo commented on ws/api/transport.go:
```diff
@@ -39,8 +39,8 @@ func MakeHandler(svc ws.Service, l log.Logger) http.Handler {
 	logger = l
 
```
also change in `endpoint.go`

dborovcanin replied:
Is this addressed?

rodneyosodo commented on bootstrap/api/transport.go:
```diff
@@ -88,13 +88,13 @@ func MakeHandler(svc bootstrap.Service, reader bootstrap.ConfigReader, logger lo
 		opts...))
 
-	r.Put("/things/state/:id", kithttp.NewServer(
+	r.Put("/things/state/:stateId", kithttp.NewServer(
```
```suggestion
	r.Put("/things/state/:thingId", kithttp.NewServer(
```

ianmuchyri replied:
Its endpoint uses the struct below
`type changeStateReq struct {
	token string
	id    string
	State bootstrap.State  }`

in  bootstrap/api/requests.go
Doesn't this suggest the id is for the state?

rodneyosodo replied:
The endpoint is to change the state of a thing. Have a look at the service definition to where that endpoint goes.

ianmuchyri replied:
okay

rodneyosodo commented on twins/api/http/transport.go:
```diff
@@ -74,7 +74,7 @@ func MakeHandler(tracer opentracing.Tracer, svc twins.Service, logger logger.Log
 	))
 
-	r.Get("/states/:id", kithttp.NewServer(
+	r.Get("/states/:stateId", kithttp.NewServer(
```
```suggestion
	r.Get("/states/:twinId", kithttp.NewServer(
```

rodneyosodo reviewed (changes requested):
Also change https://github.com/mainflux/mainflux/blob/master/auth/api/http/groups/transport.go#L67 to `groupId`

ianmuchyri committed:
Improve path parameter naming

Improve path parameter naming for:
    auth/api/http/groups/transport.go
    bootstrap/api/transport.go
    twins/api/http/transport.go
    ws/api/endpoints.go

Signed-off-by: ianmuchyri <ianmuchiri8@gmail.com>

rodneyosodo commented on api/openapi/websocket.yml:
```diff
@@ -4,7 +4,7 @@ info:
   description: WebSocket API for sending messages through communication channels.
   version: "1.0.0"
 paths:
-  /channels/{id}/messages:
```
Fix this swagger file. You can use a tool [like](https://editor.swagger.io/) this to cross check the edited swagger files

ianmuchyri replied:
Done

ianmuchyri committed:
fix swagger files

Updated the following swagger files
    api/openapi/auth.yml
    api/openapi/cert.yml
    api/openapi/websocket.yml

Signed-off-by: ianmuchyri <ianmuchiri8@gmail.com>

dborovcanin reviewed (commented):
Looks good, please resolve the remarks left b y @rodneyosodo.

rodneyosodo commented on api/openapi/websocket.yml:
```diff
@@ -4,7 +4,7 @@ info:
   description: WebSocket API for sending messages through communication channels.
```
This is an async api. I would sugggest we move it to `api/asyncapi`. You can use this [reference](https://www.asyncapi.com/) to help you rewrite it and [this](https://studio.asyncapi.com/)

ianmuchyri committed:
Move websocket.yml from openapi to asyncapi

deleted websocket.yml file in openapi and created websocket.yml file
in asyncapi

Signed-off-by: ianmuchyri <ianmuchiri8@gmail.com>

rodneyosodo commented on api/asyncapi/websocket.yml:
```diff
@@ -0,0 +1,78 @@
+      schema:
+        type: string
+        default: ''
+
```
Format new line correctly

rodneyosodo commented on api/asyncapi/websocket.yml:
```diff
@@ -0,0 +1,78 @@
+      schema:
+        type: string
+        format: uuid
+    subtopic:
```
Declare it as optional. It is currently declared as required

ianmuchyri committed:
Update websocket.yml file

Update the websocket.yml file to make subtopic optional and
added security (bearerAuth)

Signed-off-by: ianmuchyri <ianmuchiri8@gmail.com>

ianmuchyri committed:
Format new line correctly

format newline for api/asyncapi/websocket.yml

Signed-off-by: ianmuchyri <ianmuchiri8@gmail.com>

rodneyosodo reviewed (changes requested):
I am still getting errors on `api/asyncapi/websocket.yml`. You can use [this](https://studio.asyncapi.com/) to check

dborovcanin commented on api/asyncapi/websocket.yml:
```diff
@@ -0,0 +1,88 @@
+asyncapi: "2.6.0"
```
Seems that this doc is not valid when I test in Async API [editor](https://studio.asyncapi.com/#schema-jsonMsg). Please check.

dborovcanin commented on api/asyncapi/websocket.yml:
```diff
@@ -0,0 +1,88 @@
+    description: Test broker
+    variables:
+      port:
+        description: MF_WS_Adapter port
```
Mainflux WebSocket Adapter port

dborovcanin commented on api/asyncapi/websocket.yml:
```diff
@@ -0,0 +1,88 @@
+servers:
+  dev:
+    url: localhost:{port}
+    protocol: ws
```
`WebSocket`, instead of `ws`.

ianmuchyri replied:
I will check on that

ianmuchyri replied:
From the documentation in [asyncapi.com](https://www.asyncapi.com/docs/tutorials/getting-started/servers) we would use `ws` to represent WebSocket protocol.

ianmuchyri committed:
update websocket.yml file

update the websocket.yml file based on the requested review changes
    The document is now valid

Signed-off-by: ianmuchyri <ianmuchiri8@gmail.com>

rodneyosodo reviewed (changes requested):
Change to `api/openapi/consumers-notifiers.yml` L189 `example: /subscriptions/{subId}`

ianmuchyri committed:
Update path parameter naming

make changes to path parameter naming in:
    api/openapi/consumers-notifiers.yml

Signed-off-by: ianmuchyri <ianmuchiri8@gmail.com>

dborovcanin commented on auth/api/http/keys/transport.go:
```diff
@@ -34,14 +34,14 @@ func MakeHandler(svc auth.Service, mux *bone.Mux, tracer opentracing.Tracer, log
 	))
 
-	mux.Get("/keys/:id", kithttp.NewServer(
+	mux.Get("/keys/:keyId", kithttp.NewServer(
```
Let's be consistent with naming: `groupID` but `keyId` and `thingId`. Let's switch to `thingID`, `channelID`, `keyID`... since it's more idiomatic for Go.

ianmuchyri replied:
OKay, let me make the changes

ianmuchyri committed:
update path parameters naming

Update path parameters naming to be consistent with Go

Signed-off-by: ianmuchyri <ianmuchiri8@gmail.com>

dborovcanin commented on api/asyncapi/mqtt.yml:
```diff
@@ -29,10 +29,10 @@ servers:
 
 channels:
-  channels/{channelId}/messages/{subtopic}:
+  channels/{channelID}/messages/{subtopic}:
```
Let's use this opportunity to update to the latest AsyncAPI version. Please also validate the file using AsyncAPI Studio.

ianmuchyri replied:
This has been addressed

ianmuchyri committed:
update the mqtt.yml file

update the mqtt.yml file to the latest AsyncAPI version and
make changes on the security of the server

Signed-off-by: ianmuchyri <ianmuchiri8@gmail.com>

rodneyosodo commented on api/asyncapi/mqtt.yml:
```diff
@@ -1,14 +1,17 @@
 
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0
```
```suggestion
    url: 'https://github.com/mainflux/mainflux/blob/master/LICENSE'
```

rodneyosodo commented on api/asyncapi/mqtt.yml:
```diff
@@ -1,14 +1,17 @@
   description: |
     MQTT adapter provides an MQTT API for sending messages through the platform. MQTT adapter uses [mProxy](https://github.com/mainflux/mproxy) for proxying traffic between client and MQTT broker.
     Additionally, the MQTT adapter and the message broker are replicating the traffic between brokers.
 
```
Add contact information

```yaml
  contact:
    name: Mainflux Team
    url: 'https://github.com/mainflux/mainflux'
    email: info@mainflux.com
```

ianmuchyri replied:
ok. on it

ianmuchyri replied:
This has been addressed

ianmuchyri committed:
Add contact information

Signed-off-by: ianmuchyri <ianmuchiri8@gmail.com>

rodneyosodo commented on api/asyncapi/mqtt.yml:
```diff
@@ -84,6 +94,13 @@ components:
+      description: |
+        username is thing ID connected to the channel defined in the mqtt topic and 
+        password is thing key corresponding to the thing ID 
 
```
Add new line at the bottom

ianmuchyri committed:
Update api/asyncapi/mqtt.yml

Co-authored-by: b1ackd0t <blackd0t@protonmail.com>
Signed-off-by: ianmuchyri <ianmuchiri8@gmail.com>

dborovcanin commented on things/api/things/http/transport.go:
```diff
@@ -389,7 +397,47 @@ func decodeListByMetadata(_ context.Context, r *http.Request) (interface{}, erro
+
+	return req, nil
+}
+func decodeThingListByConnection(_ context.Context, r *http.Request) (interface{}, error) {
```
Add an empty line between functions.

ianmuchyri replied:
currently addressing it

ianmuchyri replied:
This is done

ianmuchyri committed:
Add an empty line between functions

Signed-off-by: ianmuchyri <ianmuchiri8@gmail.com>

drasko committed:
Merge branch 'master' into MF-1761

drasko reviewed (approved):
LGTM

drasko committed:
Merge branch 'master' into MF-1761
